MobiFlight/MidiBoard/MidiBoardManager.cs

```diff
@@ -0,0 +1,273 @@
+                    if (!string.IsNullOrEmpty(item.RelatedInputLabel))
+                    {
+                        if (item.RelatedIds.Length != item.MessageIds.Length)
+                            errorMessages.Add($"{item.Label}: Number of RelatedIds unequal to number of MessageIds");
+                        if (!item.RelatedInputLabel.Contains("%"))
+                            errorMessages.Add($"{item.Label}: RelatedInputLabel does not contain % placeholder.");
+                    }
+                }
+            }
+            return errorMessages;
+        }
+
+
+        private bool IsErrorInDefinition(MidiBoardDefinition def)
+        {
+            List<string> errorMessages = new List<string>();
+            errorMessages.AddRange(CollectErrorsInInputDefinition(def));
+            errorMessages.AddRange(CollectErrorsInOutputDefinition(def));
+            foreach (var error in errorMessages)
+            {
+                Log.Instance.log($"[{def.InstanceName}]: {error}", LogSeverity.Error);
+            }
+            return errorMessages.Count > 0;            
+        }
+
+
+        private void LoadDefinitions()
+        {
+            foreach (var definitionFile in Directory.GetFiles("MidiBoards", "*.midiboard.json"))
+            {
+                try
+                {
+                    var midiBoardDef = JsonConvert.DeserializeObject<MidiBoardDefinition>(File.ReadAllText(definitionFile));
+                    if (!IsErrorInDefinition(midiBoardDef))
+                    {
+                        Definitions.Add(midiBoardDef.InstanceName, midiBoardDef);
+                        Log.Instance.log($"Loaded midiBoard definition for {midiBoardDef.InstanceName}", LogSeverity.Info);
+                    }                    
+                }
+                catch (Exception ex)
+                {
+                    Log.Instance.log($"Unable to load {definitionFile}: {ex.Message}", LogSeverity.Error);
+                }
+            }         
+        }
+
+        public bool MidiBoardsConnected()
+        {
+            return MidiBoards.Count > 0;
+        }
+      
+        public void Startup()
+        {
+            ProcessTimer.Start();
+        }
+
+        private void ProcessTimer_Tick(object sender, EventArgs e)
+        {    
+            if (BoardsToBeRemoved.Count > 0)
+            {
+                foreach (var board in BoardsToBeRemoved)
+                {
+                    MidiBoards.Remove(board);
+                }
+                BoardsToBeRemoved.Clear();
+            }
+            foreach (var mb in MidiBoards)
+            {
+                mb.ProcessMessages();
+            }
+            UpdateOnAttachedOrRemovedMidiBoards();           
+        }
+
+        private void UpdateOnAttachedOrRemovedMidiBoards()
+        {
+            if (CheckAttachedRemovedCounter >= 40) // Check every 2 seconds  
```

Comment:
Similar comment to above. You can remove a layer of if statements by converting this to:

`if (CheckAttachedRemovedCounter < 40) return;`

Reply:
Same style discussion.

Reply:
Please do.